docker/scripts/deprovision.sh

```diff
@@ -207,28 +207,30 @@ baremetal_2a2 | baremetal_2a4 | baremetal_2a5 | baremetal_hua)
 esac
 
 # Run eclypsium
-if [[ $arch == x86_64 ]]; then
-	case "$facility" in
-	ssg-* | att-*)
-		echo "skipping eclypsium in unsupported facility"
-		;;
-	*)
-		case "$class" in
-		disabled.plan.here)
-			echo "skipping eclypsium on unsuppported plan"
+if [ -z "$ECLYPSIUM_TOKEN" ] && [ "${ECLYPSIUM_TOKEN}" != "" ]; then
```

Comment:
`if [[ -n "${ECLYPSIUM_TOKEN:-}" ]]`. You need the `${...:-}` because even with `-z` you are still using a possibly undefined variable and bash stops you before it runs the `-z` test if it is in fact undefined.

Then also the `if` conditions don't make sense as is anyway. `if empty && != ""` so will never match.

Reply:
Ya know, my 'entire life' I've thought -z meant something different than it turns out it actually does...     Changes made as suggested.

Reply:
heh